seatunnel-api/src/main/java/org/apache/seatunnel/api/configuration/util/ConfigValidator.java

```diff
@@ -45,14 +45,18 @@ public void validate(OptionRule rule) {
         List<RequiredOption> requiredOptions = rule.getRequiredOptions();
         for (RequiredOption requiredOption : requiredOptions) {
             validate(requiredOption);
-            requiredOption
-                    .getOptions()
-                    .forEach(
-                            option -> {
-                                if (SingleChoiceOption.class.isAssignableFrom(option.getClass())) {
-                                    validateSingleChoice(option);
-                                }
-                            });
+
+            for (Option<?> option : requiredOption.getOptions()) {
+                if (SingleChoiceOption.class.isAssignableFrom(option.getClass())) {
+                    // is required option and not match condition, skip validate
+                    if (isConditionOption(requiredOption)
+                            && !matchCondition(
+                                    (RequiredOption.ConditionalRequiredOptions) requiredOption)) {
+                        continue;
+                    }
+                    validateSingleChoice(option);
+                }
+            }
```

Comment:
cc @Hisoka-X

Reply:
This fix we have some options which have the same key but different excepted values, which will be set under different condition.
e.g. `compress_code` in file.